Unix/mak/rules.mak

```diff
@@ -176,6 +180,15 @@ CXXPROGFLAGS+=$(shell $(BUILDTOOL) syslibs)
 CXXPROGFLAGS+=$(OPENSSL_LIBS)
 CXXPROGFLAGS+=$(LIBPATHFLAGS)
 
+ifeq ($(ENABLE_GCOV),1)
+  CFLAGS+=-lgcov
+  CPROGFLAGS+=-lgcov
+  CSHLIBFLAGS+=-lgcov
+  CXXFLAGS+=-lgcov
+  CXXPROGFLAGS+=-lgcov
+  CXXSHLIBFLAGS+=-lgcov
+endif
```

Comment:
I don't think this is right, and would like a new iteration.

The `-l` qualifier is for linking in a library. So, if you say `-lgcov`, you're saying include library libgcov.{so|a} with this.

The `-l` qualifier is a LINKER flag, not a COMPILER flag. So it doesn't belong in CFLAGS, which is specifically for compiling. It may work (if CFLAGS is included when linking), but it's not where this belongs.

Please resolve this by putting it in the right place (should just be one or two variables), then posting a new iteration.